internal/model/model.go

```diff
@@ -0,0 +1,32 @@
+package model
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/carfloresf/the-sun-god/pkg/random"
+)
+
+type Post struct {
```

Comment:
I would suggest adding validation logic here
So the important part of the application logic is how would you validate Post before Update, or Create

I would move that logic here into a model, because that is sort of a hard business rule, so it should be part of the core of the application, and not the service itself. The service is responsible for calling validation, but the logic itself does not matter for the handler. It just needs to know if the post is valid for that rule.

Reply:
It is "kinda" in here, the validation logic is enforced by validator v10 using binding tags and BindJSON function, I was using validator in here before, but since it is already implemented inside Gin functions I wanted to keep my code a little simpler.
 But yeah it can be implemented also in here.